packages/by-name/microsoft/kata-runtime/package.nix

```diff
@@ -6,17 +6,82 @@
   fetchFromGitHub,
   yq-go,
   git,
+  applyPatches,
 }:
 
 buildGoModule (finalAttrs: {
   pname = "kata-runtime";
-  version = "3.2.0.azl5";
+  version = "3.15.0.aks0";
 
-  src = fetchFromGitHub {
-    owner = "microsoft";
-    repo = "kata-containers";
-    tag = finalAttrs.version;
-    hash = "sha256-yyqVQ8EPHbhwkm2OmmgyCAFbEca5pZRjHRmlGRv9PG0=";
+  src = applyPatches {
+    src = fetchFromGitHub {
+      owner = "microsoft";
+      repo = "kata-containers";
+      tag = finalAttrs.version;
+      hash = "sha256-nptvqKEzxlVgMRY/7GMBoU9LbuX5V0WLGXJ9C20+zAo=";
+    };
+
+    patches = [
+      # As we use a pinned version of the tardev-snapshotter per runtime version, and
+      # the tardev-snapshotter's directory has a hash suffix, we must allow multiple
+      # layer source directories. For now, match the layer-src-prefix with a regex.
+      # We could think about moving the specific path into the settings and set it
+      # to the expected value.
+      #
+      # This patch is not upstreamable.
+      ./0001-genpolicy-regex-check-contrast-specific-layer-src-pr.patch
+      # This patch builds on top of the Azure CSI patches specific to the msft
+      # version of genpolicy. Therefore, we don't attempt to upstream those changes.
+      # We can revisit this if microsoft upstreamed
+      # https://github.com/microsoft/kata-containers/pull/174
+      # TODO
```

Comment:
Leftover?

Reply:
Ah yes, there was some confusion with that patch.. But I think it's okay now.